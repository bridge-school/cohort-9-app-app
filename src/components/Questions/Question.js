import React from "react";
import { Form, Checkbox } from 'semantic-ui-react'
import "./Quetions.css";

const Question = ({
  question,
  index,
  onPromptChange,
  onOptionsChange,
  onTypeChange,
  onIsRequiredChange,
  onDelete
}) => {
  const qNumber = index + 1;
  const optionsString = question.options ? question.options.join(', ') : ''
  const QUESTION_TYPE_OPTIONS = [
    { key: 'short-answer', text: 'Short Answer', value: 'short-answer'},
    { key: 'drop-down', text: 'Dropdown', value: 'drop-down'},
    { key: 'check-box', text: 'Checkboxes', value: 'check-box'},
    { key: 'paragraph', text: 'Paragraph', value: 'paragraph'}
  ]

  return (
    <>
      <Form.Group>
        <Form.Input
          id={`q${index}__name`}
          label={{ children: `Question #${qNumber}`, htmlFor: `q${index}__name` }}
          onChange={(e, {value}) => {
            onPromptChange(index, value);
          }}
          value={question.prompt}
          width={8}
        />
        <Form.Select
          fluid
          id={`q${index}__type`}
          label={{ children: `Question #${qNumber} Type`, htmlFor: `q${index}__type` }}
          options={QUESTION_TYPE_OPTIONS}
          onChange={(e, {value}) => {
            onTypeChange(index, value);
          }}
          value={question.type}
          width={4}
        />
        <Form.Field
          id={`q${index}__req`}
          className="Question-checkbox"
          control={Checkbox}
          label={{ children: `Is Required`, htmlFor: `q${index}__req` }}
          onChange={(e, {checked: value}) => {
            onIsRequiredChange(index, value);
          }}
          checked={question.required}
          width={2}
        />
        <Form.Button
          className="Question-deleteBtn"
          content="X"
          type="button"
          onClick={() => {
            onDelete(index);
          }}
          width={2}
        />
      </Form.Group>
      {(question.type === "check-box" || question.type === "drop-down") &&
        <Form.Group>
          <Form.Input
            id={`q${index}__options`}
            label={{ children: `Question #${qNumber} type options`, htmlFor: `q${index}__options` }}
            onChange={(e, {value}) => {
              onOptionsChange(index, value);
            }}
            value={optionsString}
            placeholder="a, b, c, d"
            width={16}
          />
        </Form.Group>
      }
    </>
  );
};

export default Question;
