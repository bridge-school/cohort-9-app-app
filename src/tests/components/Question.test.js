import React from "react";
import renderer from "react-test-renderer";
import Question from "../../components/Questions/Question";

describe("Question", () => {
  test("Should render checkboxes properly", () => {
    const question = {
      timestampForKey: '1575845632',
      prompt: 'question 1',
      options: ['a', 'b'],
      type: 'check-box',
      isRequired: true,
    }
    const index = 0;
    const tree = renderer
      .create(<Question question={question} index={index} />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  test("Should render dropdown properly", () => {
    const question = {
      timestampForKey: '1575845632',
      prompt: 'question 2',
      options: ['a', 'b', 'c', 'd'],
      type: 'drop-down',
      isRequired: false
    }
    const index = 1;
    const tree = renderer
      .create(<Question question={question} index={index} />)
      .toJSON();
  
    expect(tree).toMatchSnapshot();
  });
  test("Should render paragraph properly", () => {
    const question = {
      timestampForKey: '1575845632',
      prompt: 'question 3',
      type: 'paragraph',
      isRequired: false,
    }
    const index = 2;
    const tree = renderer
      .create(<Question question={question} index={index} />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });
  test("Should render short answers properly", () => {
    const question = {
      timestampForKey: '1575845632',
      prompt: 'question 4',
      type: 'short-answer',
      isRequired: true,
    }
    const index = 3;
    const tree = renderer
      .create(<Question question={question} index={index} />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});