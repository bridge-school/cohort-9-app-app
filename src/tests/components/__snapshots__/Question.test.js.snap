// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Should render checkboxes properly 1`] = `
Array [
  <div
    className="fields"
  >
    <div
      className="eight wide field"
    >
      <label
        htmlFor="q0__name"
      >
        Question #1
      </label>
      <div
        className="ui input"
      >
        <input
          id="q0__name"
          onChange={[Function]}
          type="text"
          value="question 1"
        />
      </div>
    </div>
    <div
      className="four wide field"
    >
      <label
        htmlFor="q0__type"
      >
        Question #1 Type
      </label>
      <div
        aria-expanded={false}
        className="ui fluid selection dropdown"
        id="q0__type"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-atomic={true}
          aria-live="polite"
          className="text"
          role="alert"
        >
          Checkboxes
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Short Answer
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Dropdown
            </span>
          </div>
          <div
            aria-checked={true}
            aria-selected={true}
            className="active selected item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Checkboxes
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Paragraph
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="two wide field Question-checkbox"
    >
      <div
        className="ui checkbox"
        onChange={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          id="q0__req"
          readOnly={true}
          tabIndex={0}
          type="checkbox"
        />
        <label
          htmlFor="q0__req"
        >
          Is Required
        </label>
      </div>
    </div>
    <div
      className="two wide field Question-deleteBtn"
    >
      <button
        className="ui button"
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
  </div>,
  <div
    className="fields"
  >
    <div
      className="sixteen wide field"
    >
      <label
        htmlFor="q0__options"
      >
        Question #1 type options
      </label>
      <div
        className="ui input"
      >
        <input
          id="q0__options"
          onChange={[Function]}
          placeholder="a, b, c, d"
          type="text"
          value="a, b"
        />
      </div>
    </div>
  </div>,
]
`;

exports[`Question Should render dropdown properly 1`] = `
Array [
  <div
    className="fields"
  >
    <div
      className="eight wide field"
    >
      <label
        htmlFor="q1__name"
      >
        Question #2
      </label>
      <div
        className="ui input"
      >
        <input
          id="q1__name"
          onChange={[Function]}
          type="text"
          value="question 2"
        />
      </div>
    </div>
    <div
      className="four wide field"
    >
      <label
        htmlFor="q1__type"
      >
        Question #2 Type
      </label>
      <div
        aria-expanded={false}
        className="ui fluid selection dropdown"
        id="q1__type"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-atomic={true}
          aria-live="polite"
          className="text"
          role="alert"
        >
          Dropdown
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Short Answer
            </span>
          </div>
          <div
            aria-checked={true}
            aria-selected={true}
            className="active selected item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Dropdown
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Checkboxes
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Paragraph
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="two wide field Question-checkbox"
    >
      <div
        className="ui checkbox"
        onChange={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          id="q1__req"
          readOnly={true}
          tabIndex={0}
          type="checkbox"
        />
        <label
          htmlFor="q1__req"
        >
          Is Required
        </label>
      </div>
    </div>
    <div
      className="two wide field Question-deleteBtn"
    >
      <button
        className="ui button"
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
  </div>,
  <div
    className="fields"
  >
    <div
      className="sixteen wide field"
    >
      <label
        htmlFor="q1__options"
      >
        Question #2 type options
      </label>
      <div
        className="ui input"
      >
        <input
          id="q1__options"
          onChange={[Function]}
          placeholder="a, b, c, d"
          type="text"
          value="a, b, c, d"
        />
      </div>
    </div>
  </div>,
]
`;

exports[`Question Should render paragraph properly 1`] = `
<div
  className="fields"
>
  <div
    className="eight wide field"
  >
    <label
      htmlFor="q2__name"
    >
      Question #3
    </label>
    <div
      className="ui input"
    >
      <input
        id="q2__name"
        onChange={[Function]}
        type="text"
        value="question 3"
      />
    </div>
  </div>
  <div
    className="four wide field"
  >
    <label
      htmlFor="q2__type"
    >
      Question #3 Type
    </label>
    <div
      aria-expanded={false}
      className="ui fluid selection dropdown"
      id="q2__type"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-atomic={true}
        aria-live="polite"
        className="text"
        role="alert"
      >
        Paragraph
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Short Answer
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Dropdown
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Checkboxes
          </span>
        </div>
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Paragraph
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="two wide field Question-checkbox"
  >
    <div
      className="ui checkbox"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        id="q2__req"
        readOnly={true}
        tabIndex={0}
        type="checkbox"
      />
      <label
        htmlFor="q2__req"
      >
        Is Required
      </label>
    </div>
  </div>
  <div
    className="two wide field Question-deleteBtn"
  >
    <button
      className="ui button"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>
</div>
`;

exports[`Question Should render short answers properly 1`] = `
<div
  className="fields"
>
  <div
    className="eight wide field"
  >
    <label
      htmlFor="q3__name"
    >
      Question #4
    </label>
    <div
      className="ui input"
    >
      <input
        id="q3__name"
        onChange={[Function]}
        type="text"
        value="question 4"
      />
    </div>
  </div>
  <div
    className="four wide field"
  >
    <label
      htmlFor="q3__type"
    >
      Question #4 Type
    </label>
    <div
      aria-expanded={false}
      className="ui fluid selection dropdown"
      id="q3__type"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-atomic={true}
        aria-live="polite"
        className="text"
        role="alert"
      >
        Short Answer
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Short Answer
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Dropdown
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Checkboxes
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Paragraph
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="two wide field Question-checkbox"
  >
    <div
      className="ui checkbox"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        id="q3__req"
        readOnly={true}
        tabIndex={0}
        type="checkbox"
      />
      <label
        htmlFor="q3__req"
      >
        Is Required
      </label>
    </div>
  </div>
  <div
    className="two wide field Question-deleteBtn"
  >
    <button
      className="ui button"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>
</div>
`;
